src/rv/comm/rcssserver/GameState.java

```diff
@@ -48,6 +48,34 @@
         void gsServerMessageProcessed(GameState gs);
     }
 
+    public enum FoulType {
+        CROWDING(0, "crowding"), TOUCHING(1, "touching"), ILLEGALDEFENCE(2, "illegal defence"),
```

Comment:
I would prefer to use `UPPER_CASE` instead of `UPPERCASE` to stay consistent with the rest of the codebase (i.e. `ILLEGAL_DEFENCE`, `ILLEGAL_ATTACK`).